src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -1479,6 +1479,24 @@ public List<PartitionPositionBounds> getPositionsForRanges(Collection<Range<Toke
         return positions;
     }
 
+    public long onDiskSizeForRanges(Collection<Range<Token>> ranges)
+    {
+        long total = 0;
+        if (!compression)
+        {
+            for (var position : getPositionsForRanges(ranges))
+                total += position.upperPosition - position.lowerPosition;
+        }
+        else
+        {
+            final CompressionMetadata compressionMetadata = getCompressionMetadata();
```

Comment:
are we exercising this branch in the unit tests ?

Reply:
There's a list of TODOs in `UnifiedCompactionStrategy.createAndAddTasks` where this is next in line.